src/utils/cache.ts

```diff
@@ -0,0 +1,35 @@
+import type { Stats } from 'fs'
+
+export type FileCache = Map<string, Promise<string>>
+export type LstatCache = Map<string, Promise<Stats>>
+export type ReaddirCache = Map<string, Promise<string[]>>
+
+interface NFTCache {
+  fileCache: FileCache
+  statCache: Map<string, unknown>
+  symlinkCache: Map<string, unknown>
+  analysisCache: Map<string, unknown>
+}
+
+export interface RuntimeCache {
+  // file content
+  fileCache: FileCache
+  lstatCache: LstatCache
+  readDirCache: ReaddirCache
+  // NFT cache, which should not be used in zisi and only supplied to NFT
+  // this cache shares the file cache with zisi
+  nftCache: Partial<NFTCache>
+}
+
+export const createNewCache = (): RuntimeCache => {
```

Comment:
I find it curious that you have a method called `createNew<something>` that returns an object with internal state. Any reason not to use a class and do `new Cache()`?

Reply:
Huh, I didn't even think about this. Yes that makes sense. Done